src/main/java/gregtech/common/metatileentities/multi/electric/generator/MetaTileEntityLargeTurbine.java

```diff
@@ -100,7 +107,21 @@ protected void addDisplayText(List<ITextComponent> textList) {
             FluidStack fuelStack = ((LargeTurbineWorkableHandler) recipeMapWorkable).getInputFluidStack();
             int fuelAmount = fuelStack == null ? 0 : fuelStack.amount;
 
+            // energy output (recipeMapWorkable.getRecipeEUt())
+            // recipe fluid (recipeMapWorkable.getPreviousRecipe().fluidInputs)
+            // recipe eut (recipeMapWorkable.getPreviousRecipe().EUt)
+            // recipe duration (recipeMapWorkable.getPreviousRecipe().duration)
+            // recipeMapWorkable.getPreviousRecipe().EUt * recipeMapWorkable.getPreviousRecipe().duration / recipeMapWorkable.getPreviousRecipe().fluidInputs.get(0).amount
             ITextComponent fuelName = new TextComponentTranslation(fuelAmount == 0 ? "gregtech.fluid.empty" : fuelStack.getUnlocalizedName());
+
+            Recipe recipe = recipeMapWorkable.getPreviousRecipe();
+            if (recipe != null) {
+                FluidStack recipeFuel = recipe.getFluidInputs().get(0).getInputFluidStack();
+                int fuelEnergyValue = recipeFuel.amount / -(recipe.getEUt());
+                int requiredFluidAmount = (int) Math.floor(fuelEnergyValue * (((LargeTurbineWorkableHandler) recipeMapWorkable).getMaxVoltage() / (rotorHolder.getTotalEfficiency() / 100f)));
```

Comment:
It would be good to use the calculation directly from the large turbine workable instead of copy-pasting parts of it if possible. This would help improve our ability for future modification.

Reply:
I couldn't find a way to get the information I needed from the workable, is there something I'm missing? Does the workable already have the L/t required to run the recipe stored somewhere?

Reply:
The parallel amount is calculated in `LargeTurbineWorkableHandler` on line `89`. This calculation could be moved to its own method, and then called from this method (`addDisplayText`). The value multiplied with the recipe's first fluid input's amount should be the fuel to consume.

Reply:
I think this is done now, let me know if the changes in 7dd06a6 and f1f64ee are satisfactory.